source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
++--------------------+----------------------------------------------------------+
+|Boolean             |*true* or *false*                                         |
++--------------------+----------------------------------------------------------+
+|Null                |*null*                                                    |
++--------------------+----------------------------------------------------------+
+
+.. [#] A *hex string* is a JSON string that contains only hexadecimal
+                digits [0-9a-f] (uppercase is not canonical)
+.. [#] This MUST conform to the `Decimal128 specification`_
+
+.. [#] BSON Regex options MUST be in alphabetical order.
+
+.. [#] See https://docs.mongodb.com/manual/reference/glossary/#term-namespace
+
+.. [#] See https://docs.mongodb.com/manual/reference/database-references/#dbrefs
+
+.. _Decimal128 specification: https://github.com/mongodb/specifications/blob/master/source/bson-decimal128/decimal128.rst#writing-to-extended-json
+
+Representation of Special Numeric Values
+----------------------------------------
+
+Following the `extended JSON format for the Decimal128 type`_, special numeric
+values are encoded as follows:
+
++----------------------------------------+----------------------------------------+
+|**Value**                               |**String**                              |
++----------------------------------------+----------------------------------------+
+|Positive Infinity                       |``Infinity``                            |
++----------------------------------------+----------------------------------------+
+|Negative Infinity                       |``-Infinity``                           |
++----------------------------------------+----------------------------------------+
+|NaN (all variants)                      |``NaN``                                 |
++----------------------------------------+----------------------------------------+
+
+.. _extended JSON format for the Decimal128 type: https://github.com/mongodb/specifications/blob/master/source/bson-decimal128/decimal128.rst#to-string-representation
+
+For example, a BSON floating-point number with a value of positive infinity
+would be encoded as extended JSON as follows::
+
+  {"$numberDouble": "NaN"}
+
+Option Names
+------------
+
+Parsers and generators MAY be configurable to handle both Canonical and
+Historical Extended JSON formats. In such a case, the following option names MAY
+be used to distinguish between these two formats:
+
+* ``canonicalExtendedJSON`` (references Extended JSON as described in this specification)
+* ``legacyExtendedJSON`` (references Historical Extended JSON)
+
+Deviating from these names is perfectly acceptable as described in similar
+specifications. These names may also be used as "base names" by appending or
+prepending as appropriate to form method names and other option names. This
+specification does not require any specific options to be made available.
+
+Example
```

Comment:
This doesn't seem to be example for `Option Names`. This should likely be called `Examples` or even `Extended JSON examples`